### PR TITLE
Add performance instrumentation to TelemetryManager events

### PR DESCRIPTION
**Pull request by jasonnutter**
This PR instruments all TelemetryManager events with the measurements via the `performance` API, which will put the measurements inside reports generated by the Chromium performance profiler.

Annotation 2020-05-12 112615

Adds a label to each event, as well as a bit of other refactoring.

### CODE REVIEW

File: samples/react-sample-app/src/auth-utils.js [hunk 1]
```diff
@@ -51,20 +51,24 @@ export const GRAPH_REQUESTS = {
...
...c"],"jwks_uri":"https://login.microsoftonline.com/common/discovery/v2.0/keys","response_modes_supported":["query","fragment","form_post"],"subject_types_supported":["pairwise"],"id_token_signing_alg_values_supported":["RS256"],"response_types_supported":["code","id_token","code id_token","id_token token"],"scopes_supported":["openid","profile","email","offline_access"],"issuer":"https://login.microsoftonline.com/{tenantid}/v2.0","request_uri_parameter_supported":false,"userinfo_endpoint":"https://graph.microsoft.com/oidc/userinfo","authorization_endpoint":"https://login.microsoftonline.com/common/oauth2/v2.0/authorize","http_logout_supported":true,"frontchannel_logout_supported":true,"end_session_endpoint":"https://login.microsoftonline.com/common/oauth2/v2.0/logout","claims_supported":["sub","iss","cloud_instance_name","cloud_instance_host_name","cloud_graph_host_name","msgraph_host","aud","exp","iat","auth_time","acr","nonce","preferred_username","name","tid","ver","at_hash","c_hash","email"],"tenant_region_scope":null,"cloud_instance_name":"microsoftonline.com","cloud_graph_host_name":"graph.windows.net","msgraph_host":"graph.microsoft.com","rbac_url":"https://pas.windows.net"}'
```
Comment (jasonnutter) [inline samples/react-sample-app/src/auth-utils.js:71]: Will remove.

File: samples/react-sample-app/src/auth-utils.js [hunk 2]
```diff
@@ -51,20 +51,24 @@ export const GRAPH_REQUESTS = {
...
             GRAPH_SCOPES.PROFILE,
             GRAPH_SCOPES.USER_READ
-        ]
+        ],
+        forceRefresh: true
     },
     EMAIL: {
-        scopes: [GRAPH_SCOPES.MAIL_READ]
+        scopes: [GRAPH_SCOPES.MAIL_READ],
+        forceRefresh: true
```
Comment (jasonnutter) [inline samples/react-sample-app/src/auth-utils.js:59]: Will remove.

**Review by pkanher617 — APPROVED** [review]

File: lib/msal-core/src/telemetry/TelemetryEvent.ts [hunk 3]
```diff
@@ -32,6 +31,24 @@ export default class TelemetryEvent {
     public stop(): void {
         // Set duration of event
         this.setElapsedTime(+Date.now() - +this.startTimestamp);
+
+        if ("performance" in window && window.performance.mark && window.performance.measure) {
```
Comment (DarylThayil) [inline lib/msal-core/src/telemetry/TelemetryEvent.ts:35]: Can this boolean check be factored out since you do this check again?

**Review by DarylThayil — APPROVED** [review]
one small note but lgtm

File: lib/msal-core/src/telemetry/TelemetryEvent.ts (hunk 3 above)
Comment (jasonnutter) [reply lib/msal-core/src/telemetry/TelemetryEvent.ts:35]: Yeah, thats a good call.

Comment (jasonnutter) [reply lib/msal-core/src/telemetry/TelemetryEvent.ts]: Done.

**Pull request merged**